src/components/Calculator.js

```diff
@@ -3,45 +3,63 @@
 /* eslint-disable react/prefer-stateless-function */
 import React from 'react';
 import './Calculator.css';
+import calculate from '../logic/calculate';
 
 class Calculator extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      total: 0,
+      next: 0,
+    };
+
+    this.clickHandler = this.clickHandler.bind(this);
+  }
+
+  clickHandler(e) {
+    this.setState((previousState) => calculate(previousState, e.target.innerText));
   }
 
   render() {
+    let val = 0;
+    const { total, next } = this.state;
+    if (next) {
+      val = next;
+    } else if (total) {
+      val = total;
+    }
```

Comment:
- [ ] According to the project requirements, you should have at least one lifecycle method. I would suggest you extract this code into a lifecycle method, like `componentDidUpdate()`. You can use a different method for a different purpose though provided you use a lifecycle method.

Reply:
Thank you. I used the render() method which is also part of the lifecycle

Reply:
Oh yeah. You are right.